### PR TITLE
Save/Favourite Jobs feature and other bug fixes

### PR DESCRIPTION
**Pull request by surajgovardhangaikwad**
1. Update job details page to allow a candidate to save/favourite jobs
2. Update job listing view to show star for favorited jobs
3. Update jobs listing to filter based on saved jobs
4. Same changes for advanced jobs search
5. Add favorites link in candidate dashboard
6. Update style for Applications listing for buttons
7. Fix bug in advanced job search which was redirecting the user to regular job search on search click
8. Updated styling for jobs search
9. Add check for saving jobs, only candidate is allowed to save jobs

### CODE REVIEW

File: dashboard/static/dashboard/css/dashboard.css [hunk 1]
```diff
@@ -91,7 +91,7 @@ h2 {
 .dashboard-container {
     justify-content: center;
     padding: 20px;
-    width: 1000em;
```
Comment (jackxujh) [inline dashboard/static/dashboard/css/dashboard.css]: that is a crazy number 🤯

Comment (surajgovardhangaikwad) [reply dashboard/static/dashboard/css/dashboard.css]: That is why I have removed it :P

File: dashboard/templates/dashboard/applications_list.html [hunk 2]
```diff
@@ -84,6 +84,8 @@ <h5 class="card-title align-content-center">{{ application.job.business_title }}
                                             ACCEPTED
                                         {% elif application.status == "AP" %}
                                             PENDING
+                                        {% elif application.status == "W" %}
```
Comment (jackxujh) [inline dashboard/templates/dashboard/applications_list.html]: @surajgovardhangaikwad I feel Daisy's switched to reference constants.

File: jobs/templates/jobs/job_detail.html [hunk 3]
```diff
@@ -10,12 +10,30 @@
     <link rel="stylesheet" type="text/css" href="{% static "jobs/css/job_detail.css" %}">
 
     <div class="col-md-12 mb-3 mb-md-0">
-        <div class="card py-4 h-100">
+        <div class="card py-4 h-100" style="padding-top: 0px !important;">
```
Comment (jackxujh) [inline jobs/templates/jobs/job_detail.html]: actually, `!important` is not necessary if declaring a style in the node.

File: jobs/templates/jobs/job_detail.html [hunk 4]
```diff
@@ -10,12 +10,30 @@
...
+                        </form>
+                    {% endif %}
+                    </div>
+                    <div>
+
+                    </div>
+                </div>
 
-                <div class="job-description-flex">
+                <div class="job-description-flex" style="width: 100%">
```
Comment (jackxujh) [inline jobs/templates/jobs/job_detail.html]: is it not 100% before?

**Pull request merged**